cmake/SetupOptions.cmake

```diff
@@ -51,3 +51,9 @@ if(SINGULARITY_USE_CUDA)
     message(FATAL_ERROR "Cuda without kokkos is not currently supported")
   endif()
 endif()
+
+if(NOT SINGULARITY_USE_KOKKOS)
+  message(FATAL_ERROR
+    "For unknown reasons PORTABILITY_STRATEGY_KOKKOS will always be provided to ports-of-call. "
+    "For now you must use Kokkos with singularity-opac.")
+endif()
```

Comment:
I'm stumped by this one right now... At the end of my cmake I can't detect that `PORTABILITY_STRATEGY_KOKKOS` is set anywhere, but during `make` there's always a `-DPORTABILITY_STRATEGY_KOKKOS` which breaks compilation when `PORTABILITY_STRATEGY_NONE` is used. I'm inclined to punt on this for the purposes of this PR.

Reply:
I am going to try and look at this this afternoon. If I can figure it out by COB we might want to include my fix. Otherwise yeah we can punt.

Reply:
Interesting data point is that the CI apparently can build the code just fine with `PORTABILITY_STRATEGY_NONE`. It's possible this is an error on my platform somehow. But it would be nice to understand why this is occurring.

Reply:
I believe I fixed it. Can build on my laptop with and without kokkos. There were a few little things:
- There was an order error where options were being set AFTER dependencies. 
- SINGULARITY_KOKKOS_IN_TREE did nothing. I re-enabled it.
- If you want to disable cmake finding your pre-installed libraries, set
```C++
  set(CMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY ON CACHE BOOL "" FORCE)
  set(CMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY ON CACHE BOOL "" FORCE)
```
which are now enabled when you build with `SINGULARITY_KOKKOS_IN_TREE=ON`.

Reply:
Thanks @Yurlungur! This does indeed fix compilation for me.